### PR TITLE
rework rhai config script, add new functionally

### PR DESCRIPTION
**Pull request by dragonnn**
This PR reworks rhai config scripts:
- callbacks now are using FnPtr instead of calling them by the fn string name, this fixes capturing variables from scope above the callback
- add register_capture with allows to capture any keyboard numeric/alphabetical/fn keys and pass it as an argument to registered callback
- add basic event loop implementation with allows to send events from rhai scripts to affect the window manager state
- add add_timeout function to perform delayed or repeated tasks from rhai scripts
- introduce focused_window into Anodium state and allow access to it from rhai scripts
- introduce anodize global const in rhai scripts for holding most of functions affecting the window manager and getting information from it.

### CODE REVIEW

File: src/config/eventloop.rs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+use rhai::{plugin::*, Scope};
+use rhai::{Dynamic, EvalAltResult, FnPtr};
```
Comment (PolyMeilex) [inline src/config/eventloop.rs]: Those imports are unused, is this intentional?

File: src/config/keyboard.rs [hunk 2]
```diff
@@ -1,42 +1,52 @@
+use std::cell::RefCell;
 use std::collections::HashMap;
 use std::collections::HashSet;
 use std::hash::Hash;
-use std::sync::Mutex;
+use std::rc::Rc;
 
 use rhai::plugin::*;
+use rhai::Scope;
```
Comment (PolyMeilex) [inline src/config/keyboard.rs]: Unused?

File: src/config/mod.rs [hunk 3]
```diff
@@ -1,19 +1,64 @@
...
+use smithay::backend::input::KeyState;
+use smithay::reexports::{calloop::channel::Sender, drm};
 
-use self::output::Mode;
-use self::state::StateConfig;
+use self::anodize::Anodize;
+use self::{
+    eventloop::{ConfigEvent, EventLoop},
+    output::Mode,
+};
```
Comment (PolyMeilex) [inline src/config/mod.rs]: `FuncArgs` unused 
`TryInto` unused 
`EventLoop` unused

File: src/input_handler.rs [hunk 4]
```diff
@@ -10,7 +10,9 @@ use smithay::{
         },
         session::Session,
     },
-    reexports::wayland_server::protocol::wl_pointer,
+    reexports::{
+        wayland_server::protocol::wl_pointer, winit::platform::unix::x11::ffi::NoEventMask,
+    },
```
Comment (PolyMeilex) [inline src/input_handler.rs]: Introduces new unused import `NoEventMask`

**Review by PolyMeilex — CHANGES_REQUESTED** [review]
Besides the leftover imports, looks great!

File: src/config/eventloop.rs (hunk 1 above)
Comment (dragonnn) [reply src/config/eventloop.rs]: No, just copy and paste too much, sorry forgot to clean all the stuff. Should be now fine with the last commit :).

**Comment by dragonnn** [comment]
Should be now fine :), a lite bit too much copy and paste in imports and forgot to look at them, sorry about that.

**Comment by PolyMeilex** [comment]
Thanks :tada:

**Pull request merged**